src/features/Sync/SyncSlice.ts

```diff
@@ -18,6 +91,51 @@ const { actions: SyncAction, reducer: SyncReducer } = createSlice({
     name: REDUCER.SYNC
 });
 
+const getSliceState = ({ sync }: RootState): SyncSliceState => {
+    return sync;
+};
+
+const getUrl = (state: RootState): string => {
+    const { url } = getSliceState(state);
+    return url;
+};
+
+const getUsername = (state: RootState): string => {
+    const { username } = getSliceState(state);
+    return username;
+}
+
+const getPassword = (state: RootState): string => {
+    const { password } = getSliceState(state);
+    return password;
+}
+
+const SyncSelector = {
+    getUrl,
+    getUsername,
+    getPassword
+};
+
+function* updateUrl({ payload: { url }}: UpdateUrlAction): SagaIterator {
+    const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
+    const syncOutManager = yield call(DependencyLocator.get, DEPENDENCY.SYNC_OUT_MANAGER);
+    syncOutManager.setUrl(url);
```

Comment:
personally think passing the manager these values is easier then keeping 3 sources of the values - two in memory and one in persistant storage

Reply:
Ah, I was thinking about separation of front-end (display values in UI) from back-end (use values for syncing) state (but I did also have a concern about maintaining multiple sources-of-truth). For displaying the values in the UI, just get them from the manager?

Reply:
I wouldv'e thought read from the store for ui and potentially db for backend

Reply:
Resolved by https://github.com/openmsupply/msupply-cold-chain/pull/54.